### PR TITLE
Concordia University - Refactoring Submission

### PR DESCRIPTION
**Pull request by jeremy091**
The complete set of QKSMS refactoring done as part of an academic project, by students from Concordia University.
Changes are as follows:
1. Replaced telescoping constructors in mmssms::Message with the Fluent Builder design pattern, and replaced new instance creation where appropriate.
2. Merged ContactHelper functionality into Contact as part of responsibility re-assignment. Replaced calls to ContactHelper methods with calls to appropriate Contact methods.
3. Removed duplicated code by merging Conversation.get(context, allowQuery) and Conversation.get(threadID) methods into a single method taking all 3 parameters, and renamed it to getConversation(context, threadID, allowQuery) to better reflect it's function.
4. Made ConversationQueryHandler a separate class, instead of an inner class of Conversation.

### CODE REVIEW

File: QKSMS/src/main/java/org/w3c/dom/smil/ElementExclusiveTimeContainer.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public void setEndSync(String endSync)
                                      throws DOMException;
 
     /**
-     *  This should support another method to get the ordered collection of 
+     *  This should support another method to getConversation the ordered collection of
```
Comment (moezbhatti) [inline QKSMS/src/main/java/org/w3c/dom/smil/ElementExclusiveTimeContainer.java]: Seems to be a lot of comments with the word "get" that got replaced with "getConversation"

Comment (jeremy091) [inline QKSMS/src/main/java/org/w3c/dom/smil/ElementExclusiveTimeContainer.java]: Ah didn't notice that. Reject the pull and I'll fix that up

**Comment by moezbhatti** [comment]
You should be able to just make the changes without having to close/reopen the pull request, no?

**Comment by jeremy091** [comment]
I'm not 100% familiar with git, if I make a new commit doesn't that
invalidate the request?

**Comment by moezbhatti** [comment]
Nope, it'll just update the request

**Comment by moezbhatti** [comment]
Closing for inactivity

**Comment by moezbhatti** [comment]
If you want to have this merged in, let me know and I can create a separate branch to merge it into, and then I'll clean it up myself

**Pull request closed without merging**